sympy/core/tests/test_assumptions.py

```diff
@@ -481,6 +477,9 @@ def test_composite():
     assert S(2).is_composite is False
     assert S(17).is_composite is False
     assert S(4).is_composite is True
+    x = Dummy(integer=True, positive=True, prime=False)
+    assert x.is_composite
```

Comment:
> you shouldn't silently ignore

Sorry to have missed this. After opening a PR I look for feedback there. I don't remember all the reasons that this was committed then, but Aaron (who also commented on the PR but after the commit -- I didn't see it) has made the correction and I have committed it.